src/utilities/bcl/BCLMeasure.cpp

```diff
@@ -46,52 +47,139 @@
+
+/***************
+*  R U L E S  *
+***************/
+
+static constexpr std::array<std::pair<std::string_view, std::string_view>, 4> rootToUsageTypeMap{{
+  // fileName, usageType
+  {"measure.rb", "script"},
+  {"LICENSE.md", "license"},
+  {"README.md", "readme"},
+  {"README.md.erb", "readmeerb"}
+  // ".gitkeep"      // assuming .gitkeep outside a subfolder makes zero sense...
+  // "measure.xml"   // not included in itself!
+}};
+
+static constexpr std::array<std::pair<std::string_view, std::string_view>, 3> approvedSubFolderToUsageMap{{
+  {"docs", "doc"},
+  {"tests", "test"},
+  {"resources", "resource"},
+}};
+
+static constexpr std::array<std::string_view, 1> ignoredSubFolders{"tests/output"};
+
+// TODO: do we want to keep ignoring the docs/ directory?
+static constexpr std::array<std::string_view, 1> usageTypesIgnoredOnClone{"doc"};
+
+/******************************************************************
+*  C A L C U L A T E D    T H I N G S    A N D    H E L P E R S  *
+******************************************************************/
+
+static constexpr std::array<std::string_view, rootToUsageTypeMap.size()> approvedRootFiles = []() {
+  std::array<std::string_view, rootToUsageTypeMap.size()> result;
+  for (size_t i = 0; i < rootToUsageTypeMap.size(); ++i) {
+    result[i] = rootToUsageTypeMap[i].first;
   }
+  return result;
+}();
+
+static constexpr std::array<std::string_view, approvedSubFolderToUsageMap.size()> approvedSubFolders = []() {
+  std::array<std::string_view, approvedSubFolderToUsageMap.size()> result;
+  for (size_t i = 0; i < approvedSubFolderToUsageMap.size(); ++i) {
+    result[i] = approvedSubFolderToUsageMap[i].first;
+  }
+  return result;
+}();
+
+bool BCLMeasure::isIgnoredFileName(const std::string& fileName) {
+  return (fileName.empty() || (boost::starts_with(fileName, ".") && (fileName != ".gitkeep")));
 }
 
-bool BCLMeasure::copyDirectory(const path& source, const path& destination) const {
+bool BCLMeasure::isApprovedFile(const openstudio::path& absoluteFilePath, const openstudio::path& measureDir) {
 
-  if (!openstudio::filesystem::create_directories(destination)) {
+  std::string fileName = toString(absoluteFilePath.filename());
+  if (isIgnoredFileName(fileName)) {
     return false;
   }
 
-  openstudio::path xmlPath = openstudio::filesystem::system_complete(m_bclXML.path());
+  openstudio::path parentPath = absoluteFilePath.parent_path();
 
-  for (const auto& path : openstudio::filesystem::directory_files(source)) {
-    const auto srcItemPath = source / path;
-    const auto dstItemPath = destination / path;
-
-    if (m_bclXML.hasFile(srcItemPath) || (xmlPath == openstudio::filesystem::system_complete(srcItemPath))) {
-      if (!openstudio::filesystem::copy_file_no_throw(srcItemPath, dstItemPath)) {
-        return false;
+  if (parentPath == measureDir) {
```

Comment:
This is where the unit test is failing @tijcolem .

One again I got bit by the fact that `toPath("/path/to/dir/") != toPath("/path/to/dir")` in boost::filesystem::path...

Reply:
(Path handling is absolutely infuriating in C++, whether that's boost::filesystem or std::filesystem)

Reply:
https://github.com/NREL/OpenStudio/pull/4417/commits/5eff870ab4fb71152a81c2e386773aa4570f17bf should have fixed this.